### PR TITLE
[DD4hep] cleanup file names

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

addresses issue #33373 and comments on https://github.com/cms-sw/cmssw/pull/36004

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
please test

**Comment by civanch** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by tvami** [comment]
+1
 * tests pass nicely 
 * at one point it would be nice if @cms-sw/geometry-l2 would fix the reports from https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a091d8/20303/llvm-analysis/ (there are memory leaks, ES get() calls, Division by zero, just to name a few that sounds important)

**Comment by civanch** [comment]
+1

**Comment by tvami** [comment]
urgent
 * Jenkins tests fail w/o this now

**Comment by makortel** [comment]
+core

File: Validation/HGCalValidation/test/MakePlots.C [hunk 1]
```diff
@@ -453,7 +453,7 @@ void hgcalBHValidPlots(std::string fname = "roots/hgcBHValidD83.root",
 
 void ehcalPlots(std::string fname = "ecalHitdd4hep.root",
```
Comment (srimanob) [inline Validation/HGCalValidation/test/MakePlots.C:454]: It seems the script to update does not cover "dd4hep".
For example, the filename mentioned in Validation/Geometry/test/runP_ECAL_dd4hep_cfg.py has been updated from "DD4HEP" to "DD4hep", i.e. matbdg_ECAL_DD4HEP.root, matbdg_ECAL_DD4HEP.txt

Will changing of filenames affect something else from the DPG side?

**Comment by srimanob** [comment]
+Upgrade

I made a comment on changing of filename. However, it can be clarified and fixed (if need) in the later PR.

File: Validation/HGCalValidation/test/MakePlots.C (hunk 1 above)
Comment (ianna) [reply Validation/HGCalValidation/test/MakePlots.C:454]: IMHO, usage of `dd4hep` as a namespace or as an Era is fine. So is usage of `DD4HEP` in macros or preprocessor definitions.

Other inconsistent names include `ECAL`, `ECal` and `Ecal`, etc.

**Comment by cvuosalo** [comment]
+1

I think all cases of `DD4Hep` have now been changed.

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/36021	4c995c6
- code changes are technical
    - the PR description could've been more explicit that this is a renaming to `DD4hep` (instead of just referring to other issues)
- jenkins tests pass

**Comment by qliphy** [comment]
@cms-sw/pdmv-l2   This is needed to fix IB issues  https://github.com/cms-sw/cmssw/issues/36029   Would you please have a quick check?

**Comment by kskovpen** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**